lib/http_aws_sigv4.c

```diff
@@ -497,6 +502,10 @@ static CURLcode canon_query(struct Curl_easy *data,
         }
       }
     }
+    if(!found_equals) {
+      /* queries without value still need an equals */
+      result = Curl_dyn_addn(dq, "=", 1);
```

Comment:
```suggestion
      result = Curl_dyn_addn(dq, "=", 1);
      if(result)
        return result:
```

... because otherwise it might be overwritten below and still return OK in an error situation.

Reply:
The loop checks the result every time and bails if it had failed, should we continue trying to add chars in the case of failure? If not maybe the patch should be 

```
    if(!found_equals && !result) {
      /* queries without value still need an equals */
      result = Curl_dyn_addn(dq, "=", 1);
    }
    if(i < entry - 1 && !result) {
      /* insert ampersands between query pairs */
      result = Curl_dyn_addn(dq, "&", 1);
    }
```

Reply:
Could the existing code overwrite an error from the loop in the `/* insert ampersands between query pairs */` case?

Reply:
yes, it can, so your proposed fix looks fine